app/lib/smaxtec_api.rb

```diff
@@ -5,21 +5,32 @@ class SmaxtecApi
   SMAXTEC_API_PASSWORD = Rails.application.secrets.smaxtec_api_password
   SMAXTEC_API_BASE_URL = 'https://api-staging.smaxtec.com/api/v1'
 
-  def get_temperature
+  def update_sensor_readings
+    Sensor.where.not(animal_id: [nil, ""]).each do |sensor|
+      # temperature sensor
+      if sensor.sensor_type.property == 'Temperature'
```

Comment:
This line hard-wires the code with certain records in the database. Ie. you expect data to be present in the database. This is sth. you should try to avoid. In this case, it's inevitable. But you can make the best out of it:
1. Refactor
```ruby
class Sensor::Type
  enum property: [:temperature, ...]
end

class Sensor
  delegate :temperature?, to: :sensor_type
end
```
2. Run
```
bin/rails g migration AddSensorTypes
```
3. Create the sensor types in the database (if they don't exist already)
4. Write one rspec test like:
```ruby
it 'sensor type temperature is in the database' do
  expect(Sensor::Type.temperature.count).to be > 1
end